standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -4392,19 +4392,17 @@ private List<Partition> createPartitionFolders(final List<Partition> partitionsT
   @Override
   public AddPartitionsResult add_partitions_req(AddPartitionsRequest request)
       throws TException {
-    startFunction("add_partitions_req",
-        ": db=" + request.getDbName() + " tab=" + request.getTblName());
-    AddPartitionsResult result = new AddPartitionsResult();
-    if (request.getParts().isEmpty()) {
-      return result;
-    }
     String catName = request.isSetCatName() ? request.getCatName() : getDefaultCatalog(conf);
     String dbName = request.getDbName();
     String tblName = request.getTblName();
     startTableFunction("add_partitions_req", catName, dbName, tblName);
 
+    AddPartitionsResult result = new AddPartitionsResult();
     Exception ex = null;
     try {
+      if (request.getParts().isEmpty()) {
+        return result;
```

Comment:
This is not covered in the topic of thic PR: " Fix the mismatched function names".
Why do we need this here?

Reply:
If we do not move this code into try clause, the `endFunction()` calls would be missed.